sympy/tensor/array/ndim_array.py

```diff
@@ -367,19 +367,29 @@ def __sub__(self, other):
 
     def __mul__(self, other):
         from sympy.matrices.matrices import MatrixBase
+        from sympy.tensor.array import SparseNDimArray
 
         if isinstance(other, (Iterable, NDimArray, MatrixBase)):
             raise ValueError("scalar expected, use tensorproduct(...) for tensorial product")
+
         other = sympify(other)
+        if isinstance(self, SparseNDimArray):
+            return type(self)({k: other*v for (k, v) in self._sparse_array.items()}, self.shape)
+
         result_list = [i*other for i in self]
         return type(self)(result_list, self.shape)
 
     def __rmul__(self, other):
         from sympy.matrices.matrices import MatrixBase
+        from sympy.tensor.array import SparseNDimArray
```

Comment:
Would it be possible to add this import at the top of the file as it is being used [here](https://github.com/sympy/sympy/pull/16994/files#diff-13de3b2700b5dda2cb8b559c2c56edd7R370) too? If you are facing import issues then it's fine. Just resolve this conversation in that case.

Reply:
Yes, it is possible.